MultiBuild/BuildLogic.cs

```diff
@@ -202,230 +237,503 @@ public static bool PlayerAction_Build_BuildMainLogic_Prefix(ref PlayerAction_Bui
             return false;
         }
 
-        [HarmonyPrefix, HarmonyPriority(Priority.Last), HarmonyPatch(typeof(PlayerAction_Build), "DetermineBuildPreviews")]
-        public static bool PlayerAction_Build_DetermineBuildPreviews_Prefix(ref PlayerAction_Build __instance)
+        public static PlayerAction_Build ClonePlayerAction_Build(PlayerAction_Build original)
         {
-            var runOriginal = true;
+            var nearcdClone = new NearColliderLogic();
+            nearcdClone.Init(original.nearcdLogic.planet);
+            var clone = new PlayerAction_Build()
+            {
+                factory = original.factory,
+                player = original.player,
+                nearcdLogic = nearcdClone,
+                tmpPackage = original.tmpPackage,
+                planetAux = original.planetAux,
+                cursorTarget = original.cursorTarget,
+                buildPreviews = original.buildPreviews,
+                planetPhysics = original.planetPhysics,
+                previewPose = new Pose(Vector3.zero, Quaternion.identity),
+                posePairs = new List<PlayerAction_Build.PosePair>(64),
+                startSlots = new List<PlayerAction_Build.SlotPoint>(64),
+                endSlots = new List<PlayerAction_Build.SlotPoint>(64)
+            };
+
+            return clone;
+        }
+
+        public static bool CheckBuildConditionsFast()
+        {
+            var actionBuild = GameMain.data.mainPlayer.controller.actionBuild;
 
-            if (__instance.controller.cmd.mode == 1 && __instance.player.planetData.type != EPlanetType.Gas && __instance.cursorValid)
+            var maxThreads = 8;
```

Comment:
On 8 core (hyperthreaded), it means you'd be using everything. But UI thread is waiting anyway.